src/main/java/com/ibm/cloudant/kafka/connect/CachedClientManager.java

```diff
@@ -0,0 +1,21 @@
+package com.ibm.cloudant.kafka.connect;
+
+import com.ibm.cloud.cloudant.internal.ServiceFactory;
+import com.ibm.cloud.cloudant.v1.Cloudant;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.ibm.cloudant.kafka.common.utils.JavaCloudantUtil.VERSION;
+
+public class CachedClientManager {
+
+    // Cloudant clients, keyed by connector name
+    static Map<String, Cloudant> clientCache = new ConcurrentHashMap<>();
+
+    public static Cloudant getInstance(Map<String, String> props) {
+        String connectorName = props.get("name");
```

Comment:
If we're caching only by connector name in a static class do we need to somehow handle property changes associated with that name for cases where the runtime doesn't reload this class?
This [stackoverflow](https://stackoverflow.com/questions/64782475/kafka-source-connect-job-restart-with-updated-config-but-the-task-still-used-th) seems to suggest that if you want new properties picked up in a source connector you should use a new name, but seems to imply a name change isn't necessary for a sink connector.

It seems that this does get significantly more complicated if we aren't using the entire property set to identify an instance 😞.